web/modules/custom/hoeringsportal_audit_log/src/Routing/RoutingChangeListener.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Drupal\hoeringsportal_audit_log\Routing;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\os2web_audit\Service\Logger;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\hoeringsportal_audit_log\Form\SettingsForm;
+use Drupal\node\Entity\Node;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Route change listener.
+ */
+final class RoutingChangeListener implements EventSubscriberInterface {
+
+  /**
+   * The module config.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  private $moduleConfig;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    ConfigFactoryInterface $configFactory,
+    protected RouteMatchInterface $routeMatch,
+    protected AccountInterface $currentUser,
+    #[Autowire(service: 'os2web_audit.logger')]
+    protected Logger $auditLogger,
+  ) {
+    $this->moduleConfig = $configFactory->get(SettingsForm::SETTINGS);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('current_route_match'),
+      $container->get('current_user'),
+      $container->get('os2web_audit.logger')
+    );
+  }
+
+  /**
+   * Audit on route change, by configured routes.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\ControllerEvent $event
+   *   The event to process.
+   */
+  public function auditOnRouteChange(ControllerEvent $event) {
```

Comment:
There's no route change here. Consider renaming to `onController` (which is a weird name, but tells more about what's actually going on).